File: #2_(JAVA XML){PROJECTS}/test_9_0/app/src/main/java/com/example/test_9_0/MainActivity.java
package com.example.test_9_0;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import android.content.pm.PackageManager;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

//        if (ContextCompat.checkSelfPermission(this, Manifest.permission.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION) != PackageManager.PERMISSION_GRANTED) {
//            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION}, MY_PERMISSIONS_RECORD_AUDIO);
        }
    }
//}

File: #4_(KOTLIN JETPACK_COMBOSE) {PROJECTS}/X/app/src/main/java/com/example/x/MainActivity.kt
package com.example.x

import android.content.Context
import android.content.SharedPreferences
import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Environment
import android.widget.Toast
import androidx.core.content.ContextCompat
import com.example.x.databinding.ActivityMainBinding
import java.io.File

class MainActivity : AppCompatActivity() {

    lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        addListeners()
    }

    private fun addListeners() {
        binding.save.setOnClickListener {
            val text = binding.edittext.text.toString()

            if(ContextCompat.checkSelfPermission(applicationContext, Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
                saveTextToFile(text)
            }


        }
    }

    private fun saveTextToFile(text: String) {
        val path = Environment.getExternalStorageDirectory().path
        val fileName = "sample.txt"
        val file = File("$path/$fileName")
        file.writeText(text)
    }

}